### PR TITLE
Add AbstractConnectionPainter to enable custom connections

### PR DESCRIPTION
**Pull request by paceholder**

### CODE REVIEW

File: src/DefaultConnectionPainter.cpp [hunk 1]
```diff
@@ -251,4 +199,55 @@ void ConnectionPainter::paint(QPainter *painter, ConnectionGraphicsObject const
...
 }
 
+QPainterPath DefaultConnectionPainter::getPainterStroke(ConnectionGraphicsObject const &connection) const
+{
+    auto cubic = cubicPath(connection);
+
+    QPointF const &out = connection.endPoint(PortType::Out);
+    QPainterPath result(out);
+
+    unsigned segments = 20;
```
Comment (nolankramer) [inline src/DefaultConnectionPainter.cpp:209]: This could be `constexpr`, unless you plan for this to be variable?

**Review by nolankramer — APPROVED** [review]
Changes look good so far

**Comment by paceholder** [comment]
I scoped these changes from the great pull request https://github.com/paceholder/nodeeditor/pull/386 made by @dine-j 

Needed to make a new branch because of changed file names and some fixes in the CI scripts

File: src/DefaultConnectionPainter.cpp (hunk 1 above)
Comment (paceholder) [reply src/DefaultConnectionPainter.cpp]: Fixed

**Pull request merged**